bson/_cbsonmodule.c

```diff
@@ -2478,12 +2499,15 @@ static PyObject* get_value(PyObject* self, PyObject* name, const char* buffer,
             if ((dec128 = _get_object(state->Decimal128,
                                       "bson.decimal128",
                                       "Decimal128"))) {
-                value = PyObject_CallMethod(dec128,
-                                            "from_bid",
-                                            "y#",
-                                            buffer + *position,
-                                            (Py_ssize_t)16);
+                PyObject *_bytes_obj = PyBytes_FromStringAndSize(buffer + *position, (Py_ssize_t)16);
+                if(!_bytes_obj) {
+                    goto invalid;
```

Comment:
dec128 needs to be freed before this goto.

Reply:
fixed.